src/Engine/Component/AnimationController.cpp

```diff
@@ -0,0 +1,92 @@
+#include "AnimationController.hpp"
+#include "../Animation/Skeleton.hpp"
+#include "../Animation/AnimationClip.hpp"
+#include <Utility/Log.hpp>
+#include "glm/gtc/quaternion.hpp"
+
+using namespace Component;
+
+AnimationController::AnimationController() {
+    for (unsigned int i = 0; i < 20; ++i) {
+        bones.push_back(glm::mat4(1.0f));
+    }
+}
+
+Json::Value AnimationController::Save() const {
+    Json::Value component;
+
+    if (controller != nullptr)
+        component["animationController"] = controller->path + controller->name;
+
+    if (skeleton != nullptr)
+        component["skeleton"] = skeleton->path + skeleton->name;
+
+    return component;
+}
+
+void Component::AnimationController::UpdateAnimation(float deltaTime) {
+    if (skeleton == nullptr || controller == nullptr)
+        return;
+
+    if (bones.size() != skeleton->skeletonBones.size()) {
+        bones.clear();
+        bones.shrink_to_fit();
+        bones.resize(skeleton->skeletonBones.size());
+    }
+
+    if (activeAction1 == nullptr) {
+        activeAction1 = dynamic_cast<Animation::AnimationController::AnimationAction*>(controller->animationNodes[0]);
+    }
+
+    Animation::AnimationClip::Animation * anim = activeAction1->animationClip->animation;
```

Comment:
Space before asterisk.
`activeAction1` is the result of a `dynamic_cast` and need to be checked for validity.